src/lib/__tests__/date.test.ts

```diff
@@ -1,51 +1,271 @@
+        "2021-12-05T20:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe(false)
+    })
+  })
+
+  describe("formattedOpeningHours", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("Event opens in the future", () => {
+      const period = formattedOpeningHours(
+        "2021-12-05T20:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("Opens Dec 5 at 8pm")
+    })
+
+    it("Event is running and closes in the future", () => {
+      const period = formattedOpeningHours(
+        "2017-12-05T20:00:00+00:00",
+        "2019-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("Closes Dec 30 at 5pm")
+    })
+
+    it("Event ended in the past and is closed", () => {
+      const period = formattedOpeningHours(
+        "2016-12-05T20:00:00+00:00",
+        "2016-12-30T17:00:00+00:00"
       )
+      expect(period).toBe("Closed")
+    })
+  })
+
+  describe("singleHour", () => {
+    it("includes single date in this year", () => {
+      const period = singleHour("2018-12-05T20:00:00+00:00")
+      expect(period).toBe("8:00pm")
+    })
+  })
+
+  describe("singleDate", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("includes single date in this year", () => {
+      const period = singleDate("2018-12-05T20:00:00+00:00")
+      expect(period).toBe("Dec 5 at 8:00pm")
+    })
+
+    it("includes single date in different year from now", () => {
+      const period = singleDate("2021-12-05T20:00:00+00:00")
+      expect(period).toBe("Dec 5, 2021 at 8:00pm")
+    })
+  })
+
+  describe("singleDateWithoutHour", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("includes single date in this year", () => {
+      const period = singleDateWithoutHour("2018-12-05T20:00:00+00:00")
+      expect(period).toBe("Wed, Dec 5")
+    })
+
+    it("includes single date in different year from now", () => {
+      const period = singleDateWithoutHour("2021-12-05T20:00:00+00:00")
+      expect(period).toBe("Sun, Dec 5, 2021")
+    })
+  })
+
+  describe("datesWithHours", () => {
+    const realNow = Date.now
+    beforeEach(() => {
+      Date.now = () => new Date("2018-01-30T03:24:00") as any
+    })
+    afterEach(() => {
+      Date.now = realNow
+    })
+
+    it("includes the start and end date with different years", () => {
+      const period = datesWithHours(
+        "2021-12-05T20:00:00+00:00",
+        "2022-12-30T17:00:00+00:00"
+      )
+      expect(period).toBe("Dec 5, 2021 at 8:00pm – Dec 30, 2022 at 5:00pm")
```

Comment:
Missing the TZ component https://www.notion.so/artsy/Formatting-Time-f6f4aaeeba414583a207013fec5a9794#b0cf40f292634c7b8cc2fe026089919f